torch/csrc/jit/autodiff.cpp

```diff
@@ -828,7 +826,7 @@ static ReverseDetails addReverseInline(Gradient& grad_desc) {
   const auto get_grad = [&](Value* v) -> Value* {
     auto it = grad_map.find(v);
     if (it == grad_map.end()) {
-      auto undef = graph.insertNode(graph.createUndefined());
+      auto undef = graph.insertNode(graph.createNone(TensorType::get()));
```

Comment:
I am not sure this change is correct. The grad_map previously only stored Tensor types.  I am guessing that this change would allow a Optional[Tensor] to be passed as an input for something expecting a Tensor.